### PR TITLE
Improve `loadTileJson` types

### PR DESCRIPTION
**Pull request by HarelM**
This improves the loadTileJson method types according to what's really is returned.

### CODE REVIEW

File: src/source/load_tilejson.ts [hunk 1]
```diff
@@ -7,28 +7,39 @@ import type {RequestManager} from '../util/request_manager';
 import type {TileJSON} from '../types/tilejson';
 import type {RasterDEMSourceSpecification, RasterSourceSpecification, VectorSourceSpecification} from '@maplibre/maplibre-gl-style-spec';
 
+export type LoadTileJsonResponse = {
+    tiles: Array<string>;
+    minzoom: number;
+    maxzoom: number;
+    attribution: string;
+    bounds: RasterDEMSourceSpecification['bounds'];
+    scheme: VectorSourceSpecification['scheme'];
```
Comment (neodescis) [inline src/source/load_tilejson.ts:16]: Why is the type for bounds coming from RasterDEM, while scheme is coming from Vector? I realize they're the same type, but this reads weird.

File: src/source/load_tilejson.ts [hunk 2]
```diff
@@ -7,28 +7,39 @@ import type {RequestManager} from '../util/request_manager';
...
+        const result: LoadTileJsonResponse = pick(
             // explicit source options take precedence over TileJSON
             extend(tileJSON, options),
             ['tiles', 'minzoom', 'maxzoom', 'attribution', 'bounds', 'scheme', 'tileSize', 'encoding']
         );
 
-        if (tileJSON.vector_layers) {
-            result.vectorLayers = tileJSON.vector_layers;
-            result.vectorLayerIds = result.vectorLayers.map((layer) => { return layer.id; });
+        if ('vector_layers' in tileJSON) {
```
Comment (neodescis) [inline src/source/load_tilejson.ts:41]: vector_layers could still be null/undefined, even if the property exists in tileJSON. Might want to check that too.

Comment (HarelM) [reply src/source/load_tilejson.ts:41]: Good point.

File: src/source/load_tilejson.ts (hunk 1 above)
Comment (HarelM) [reply src/source/load_tilejson.ts:16]: It's a superset, so I'll need to take something from both in any case...

Comment (HarelM) [reply src/source/load_tilejson.ts:16]: Another option is the following
```
    bounds: RasterSourceSpecification['bounds'];
    scheme: RasterSourceSpecification['scheme'];
    tileSize: number;
    encoding: RasterDEMSourceSpecification['encoding'];
```
But it's not fundamentally different...
There is no one type of source specification that has all these definitions...

File: src/source/load_tilejson.ts (hunk 2 above)
Comment (HarelM) [reply src/source/load_tilejson.ts]: Done

File: src/source/load_tilejson.ts (hunk 1 above)
Comment (neodescis) [reply src/source/load_tilejson.ts]: OK, understood. I would say go with whatever you think is easiest on the eyes.

**Review by neodescis — APPROVED** [review]

**Pull request merged**